app.rb

```diff
@@ -0,0 +1,188 @@
+      puts "[#{person.class.name}] ID: '#{person.id}', Name: #{person.name}, Age: #{person.age}"
+    end
+  end
+
+  def create_person
+    puts 'Do you want to create a student(1) or a teacher(2)? [input the number]'
+    ans = gets.chomp.to_i
+
+    puts 'Please input a number 1 for Student or 2 for Teacher.' if ans != 1 || ans != 2
+    while [1, 2].include?(ans)
+      case ans
+      when 1
+        create_student
+        break
+      when 2
+        create_teacher
+        break
+      end
+    end
+  end
+
+  def create_student
+    print 'Age: '
+    age = gets.chomp.to_i
+
+    while age < 1
+      print 'Please enter a valid age: '
+      age = gets.chomp.to_i
+    end
+
+    print 'Name: '
+    name = gets.chomp.capitalize
+    while name == ''
+      print 'Please enter a name: '
+      name = gets.chomp.capitalize
+    end
+
+    print 'Has parent permission? (Y/N) '
+    permission = gets.chomp.upcase
+
+    # rubocop:disable Lint/UnreachableLoop
+    while permission
+      case permission
+      when 'Y'
+        permission = true
+      when 'N'
+        permission = false
+      else
+        print 'Please enter Y for Yes or N for No: '
+        permission = gets.chomp.upcase
+      end
+      break
+    end
+    # rubocop:enable Lint/UnreachableLoop
+
+    student = Student.new(age, nil, name, parent_permission: permission)
+    @people.push(student)
+    puts 'Person created successfully'
+    puts
+  end
+
+  def create_teacher
+    print 'Age: '
+    age = gets.chomp.to_i
+    while age < 1
+      print 'Please enter a valid age: '
+      age = gets.chomp.to_i
+    end
+
+    print 'Name: '
+    name = gets.chomp.capitalize
+    while name == ''
+      print 'Please enter a name: '
+      name = gets.chomp.capitalize
+    end
+
+    print 'Specialization: '
+    specialization = gets.chomp
+
+    teacher = Teacher.new(age, specialization, name)
+    @people.push(teacher)
+    puts 'Person created successfully'
+    puts
+  end
+
+  def create_book
+    print 'Book title: '
+    title = gets.chomp.capitalize
+    while title == ''
+      print 'Book title: '
+      title = gets.chomp.capitalize
+    end
+
+    print 'Book author: '
+    author = gets.chomp.capitalize
+    while author == ''
+      print 'Book author: '
+      author = gets.chomp.capitalize
+    end
+
+    book = Book.new(title, author)
+    @books.push(book)
+    puts 'Book Created successfully'
+    puts
+  end
+
+  def create_rental
+    if @books.empty?
+      puts 'Sorry there are no books avalibales'
+      return
+    end
+    puts 'Select a book from the follwoing list by number'
+    @books.each_with_index { |book, index| puts "#{index}) Title: #{book.title}, Author: #{book.author}" }
+    rented_book = gets.chomp.to_i
+
+    # rubocop:disable Lint/UnreachableLoop
+    while @books[rented_book].nil?
+      print 'Please select a book by index: '
+      rented_book = gets.chomp.to_i
+      break
+    end
+    # rubocop:enable Lint/UnreachableLoop
```

Comment:
- [ ] Please try to avoid disabling linters as the help enforce good coding style which is better for you in the long run 👍